drivers/timer/intel_adsp_timer.c

```diff
@@ -214,16 +220,24 @@ void smp_timer_init(void)
 }
 
 /* Runs on core 0 only */
-static int sys_clock_driver_init(void)
+static int sys_clock_driver_init(const struct device *dev)
 {
 	uint64_t curr = count();
-
-	IRQ_CONNECT(TIMER_IRQ, 0, compare_isr, 0, 0);
+	if (!sys_clock_driver_reinit)
+		IRQ_CONNECT(TIMER_IRQ, 0, compare_isr, 0, 0);
 	set_compare(curr + CYC_PER_TICK);
 	last_count = curr;
 	irq_init();
 	return 0;
 }
 
-SYS_INIT(sys_clock_driver_init, PRE_KERNEL_2,
-	 CONFIG_SYSTEM_CLOCK_INIT_PRIORITY);
+#ifdef CONFIG_PM
```

Comment:
remove this ifdef, that is regular API for sys_clock and you are not using anything that requires `CONFIG_PM`. If this function is not called the linker you remove it. We rely on it for this sort of thing.

Reply:
Restored to previous SYS_INIT macro